### ISSUE TITLE
Auth tests are failing

### ISSUE THREAD

**Issue by xadahiya** [HTTP-APIs/hydrus#114]
### Current Behaviour:
Auth tests ([test_auth.py](https://github.com/HTTP-APIs/hydrus/blob/master/hydrus/tests/test_auth.py)) needs to be updated as all the 4 tests are currently failing.

![sketch 41](https://user-images.githubusercontent.com/9114274/36551882-a2c6cf92-181e-11e8-89c1-e9b720d8b0a4.png)

### Expected Behaviour:
All tests should pass.

### Steps to reproduce:
Run `hydrus/tests/test_auth.py`

### Do you want to work on this issue?
no

**Comment by alohamora** [comment]
@xadahiya I checked with the codebase...the assertion error is because we have set up authentication for all the endpoints in api. `with set_authentication(self.app, True):` at `line 55 in test_auth.py`.
Hence it shall always give a response with `status code 401`.
So should I remove the authentication for the tests or shall I change the assertion to
`assert response_get.status_code == 401`

**Comment by chrizandr** [comment]
Try reverting to before #88 and then running the tests.

As far as I know no major changes other than that were made.

**Comment by alohamora** [comment]
I reverted to before #88 but the tests were still failing.
But instead I changed the encoding format of the authentication header as follows:
`self.auth_header = {"Authorization": "Basic " + b64encode(b"1:test").decode("ascii")}` 
and the byte string generated int the auth header matched with that generated while requesting through a web browser i.e. `MTp0ZXN0` and then all the tests were passing.
@xadahiya @chrizandr comments on this approach??
